### PR TITLE
updating developer documentation

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -29,18 +29,73 @@ Guidelines](https://opensource.google.com/conduct/).
...
+The profiles are:
+
+* `dsv1` - Running both Scala/DSv1 connectors.
+* `dsv1_2.11` - Running just the Scala 2.11 connector.
+* `dsv1_2.12` - Running just the Scala 2.12 connector.
+* `dsv2` - Running both Java/DSv2 connectors.
+* `dsv2_2.4` - Running just the Java Spark 2.4 connector.
+* `dsv2_3` - Running just the Java Spark 3 connector.
+* `all` - Running all the connectors.
+
```
Comment (himanshukohli09) [inline CONTRIBUTING.md:87]: We can add a small example here, how to run these profiles, how to build the connector or how to create the jar. Similar to the one given below.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -29,18 +29,73 @@ Guidelines](https://opensource.google.com/conduct/).
...
+  Scala versions), using the new DataSource APIs. Still under development.
+
+The project's artifacts are:
+  
+* `spark-bigquery-parent` - The parent POM for all artifacts. Common settings
+  and artifact version should be defined here.
+* `bigquery-connector-common` - Utility classes for working with the BigQuery
+  APIs. This artifact has no dependency on Spark.
+* `spark-bigquery-connector-common`- Common code shared among all connectors
+  (Scala and Java alike).
```
Comment (himanshukohli09) [inline CONTRIBUTING.md:50]: `bigquery-connector-common`  v/s  `spark-bigquery-connector-common` looks a bit confusing. Maybe we should add a line or two.

**Review by himanshukohli09 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -29,18 +29,78 @@ Guidelines](https://opensource.google.com/conduct/).
...
+  Java only DataSource V2 implementations.
+* `spark-bigquery-dsv2/spark-bigquery-spark24` - A Java only DataSource V2
+  connector implementing the Spark 2.4 APIs.
+* `spark-bigquery-dsv2/spark-bigquery-spark3` - A Java only DataSource V2
+  connector implementing the Spark 3 APIs. Under development.
+* `spark-bigquery-python-lib` - The python support library, adding BigQuery
+  types not supported by Spark.
+
+As building and running all the connectors is a lengthy process, the project is
+split into several artifacts, and only part of them can be built at a single run
```
Comment (emkornfield) [inline CONTRIBUTING.md:78]:
```suggestion
split into several artifacts. Each can be selectively built
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -29,18 +29,78 @@ Guidelines](https://opensource.google.com/conduct/).
...
+using [Maven profiles](https://maven.apache.org/guides/introduction/introduction-to-profiles.html).
+The profiles are:
```
Comment (emkornfield) [inline CONTRIBUTING.md:80]: the preceding paragraph mentions 'artifacts' not 'profiles' should there be consistency in the language?

**Review by emkornfield — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (davidrabinowitz) [reply CONTRIBUTING.md:80]: A profile refers to several artifacts, for example `dsv1_2.12` -> `spark-bigquery-dsv1-spark3-support`, `spark-bigquery-dsv1-parent`,`spark-bigquery_2.12`, `spark-bigquery-with-dependencies-parent`, `spark-bigquery-with-dependencies_2.12`

Comment (davidrabinowitz) [reply CONTRIBUTING.md]: I've re-phrased the paragraph, PTAL

**Review by himanshukohli09 — APPROVED** [review]

**Pull request merged**